packages/ember-core/src/validation/validators/password.ts

```diff
@@ -29,56 +35,38 @@ export default class PasswordValidator<
     super(bind, options, context);
 
     if (
-      isNone(options.minLength) &&
-      isNone(options.maxLength) &&
       isNone(options.minClasses)
     ) {
       assert(
-        'PasswordValidator requires `minLength`, `maxLength`, and `minClasses` to be provided',
+        'PasswordValidator requires `minClasses` to be provided',
       );
     }
   }
 
-  validate(value: T, options: PasswordOptions): ValidateFnResponse {
-    const {minLength, maxLength, minClasses} = options;
-
-    if (minLength !== undefined && typeof value === 'string' && value.length < minLength) {
-      return {
-        key: 'nrg.validation.passwordLengthTooShort',
-        value,
-        minLength,
-      };
-    }
-
-    if (minLength !== undefined && typeof value === 'string' && value.length < minLength) {
-      return {
-        key: 'nrg.validation.passwordLengthTooLong',
-        value,
-        maxLength,
-      };
-    }
+  validate(value: string, options: PasswordOptions): ValidateFnResponse {
+    const { minClasses } = options;
 
-    const conditions = [
-      /[A-Z]/.test(value as string),
-      /[a-z]/.test(value as string),
-      /\d/.test(value as string),
-      /[^a-zA-Z0-9]/.test(value as string),
-    ];
+    const commonTests = {
+      alpha: /(?:[A-Z]|[a-z])/,
+      upper: /[A-Z]/,
+      lower: /[a-z]/,
+      numeral: /\d/,
+      special: /[@#$%^&*\-_+=[\]{}|\\:',.?/`~"();!]/,
+    };
 
-    const classCount = conditions.filter(Boolean).length;
+    const classCount = Object.keys(commonTests)
+      .map(key => commonTests[key as keyof typeof commonTests].test(value))
+      .filter(Boolean)
+      .length;
```

Comment:
These changes are good, but you aren't using the `tests` option. I first suggested the type `(string | RegExp)[]` so that any string would be a valid test type, but once I created a sample object of common tests, I narrowed the type to `(keyof CommonTests | RegExp)[]`. This type means that in addition to regular expressions, the strings `'alpha'`, `'upper'`, `'lower'`, `'numeral'`, and `'special'` are allowed, but no other strings.